src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -141,6 +146,48 @@ export function makeMutationCountVsCnaTooltip(sampleToAnalysisGroup?:{[sampleKey
     };
 }
 
+export function generateScatterPlotDownloadData(data: IStudyViewScatterPlotData[],
+                                                sampleToAnalysisGroup?: {[sampleKey:string]:string},
+                                                analysisClinicalAttribute?: ClinicalAttribute,
+                                                analysisGroups?: AnalysisGroup[])
+{
+    const header = ["Cancer Study", "Patient ID", "Sample ID", "Mutation Count", "CNA Fraction"];
+    let valueToGroup: {[value: string] : AnalysisGroup};
+
+    if (analysisClinicalAttribute !== undefined && sampleToAnalysisGroup !== undefined) {
+        header.push(analysisClinicalAttribute.displayName);
+
+        if (analysisGroups !== undefined) {
+            header.push("Color");
+            valueToGroup = _.keyBy(analysisGroups, 'value');
+        }
+    }
+
+    const rows = data.map(datum => {
+        const row = [
+            `${datum.studyId}`,
```

Comment:
curious why you are wrapping in string this way.  why not just plain variable or var.toString?

Reply:
`undefined.toString()` gives an error, but this doesn't. Though, not sure if we really want to see `undefined` as an output.